mlir/lib/Dialect/Affine/Transforms/AffineDataCopyGeneration.cpp

```diff
@@ -116,6 +116,8 @@ void AffineDataCopyGeneration::runOnBlock(Block *block,
   AffineCopyOptions copyOptions = {generateDma, slowMemorySpace,
                                    fastMemorySpace, tagMemorySpace,
                                    fastMemCapacityBytes};
+  auto &topRegion =
+      block->getParent()->getParentOfType<func::FuncOp>().getBody();
```

Comment:
`getParentOfType` can fail which needs to be handled (fallback to getParentRegion?), and we also should check for a FunctionOpInterface instead for generality.

(here and everywhere else)